joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/TypeHelper.scala

```diff
@@ -134,8 +137,10 @@ trait TypeHelper { this: AstCreator =>
       case Some(tpe) => Seq(tpe)
       case None if node.isDefined =>
         typeFor(node.get) match {
-          case t if t != Defines.Any && t != "this" => Seq(t)
-          case _                                    => rootTypeDecl.map(_.fullName).toSeq
+          case tpe if tpe != Defines.Any && tpe != "this" =>
+            val typeFullName = if (Defines.isBuiltinType(tpe)) tpe else Defines.Any
```

Comment:
since this is used for a dynamic type hint, we shouldn't put `ANY` there. Dynamic type hints kill all further type resolution, so if we're unsure about the type (or can't express it), we shouldn't put *something* there but leave it empty instead.